packages/core/package-manager/src/NodePackageManager.js

```diff
@@ -51,7 +51,7 @@ export class NodePackageManager implements PackageManager {
   }
 
   load(resolved: FilePath, from: FilePath) {
-    if (!path.isAbsolute(resolved)) {
+    if (!path.isAbsolute(resolved) || process.versions.pnp != null) {
```

Comment:
Interesting. I'm not sure, but we'd probably want it to go through our FileSystem too. For example, if you're using a MemoryFS or some kind of OverlayFS, the normal NodePackageManager will handle this and resolve modules in those file systems instead of the native file system. I'm not familiar enough with how PnP works, but I'm guessing it just uses native FS...

Reply:
> but I'm guessing it just uses native FS...

No, with the current pre-release version of Yarn 2 ("Berry"), the modules resolve to something like `my-project/.yarn/cache/lodash-es@npm:6.1.3.zip/lodash.js`.
It patches the FS to make this transparent to e.g. Parcel without ever actually extracing that zip file to the FS. (e.g. https://github.com/yarnpkg/berry/blob/7244212072e3ac9ce9c90aadf3b2c8e681ea949c/packages/yarnpkg-fslib/sources/index.ts)

Reply:
Right, but that path is still on the native filesystem. If someone had `node_modules/foo` in their MemoryFS instance, this wouldn't pick that up. Same for relative paths, which I assume pnp just falls back to native node resolution for.